homeassistant/components/nasweb/coordinator.py

```diff
@@ -0,0 +1,132 @@
+"""Message routing coordinators for handling NASweb push notifications."""
+import asyncio
+from datetime import timedelta
+import logging
+import time
+from typing import Any, Optional
+
+from aiohttp import web
+from webio_api import Output as NASwebOutput, WebioAPI
+from webio_api.const import KEY_DEVICE_SERIAL, KEY_OUTPUTS, KEY_TYPE, TYPE_STATUS_UPDATE
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import STATUS_UPDATE_MAX_TIME_INTERVAL
+from .relay_switch import RelaySwitch
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class NotificationCoordinator:
+    """Coordinator redirecting push notifications for this integration to appropriate NASwebCoordinator."""
+
+    def __init__(self) -> None:
+        """Initialize coordinator."""
+        self._coordinators: dict[str, NASwebCoordinator] = {}
+
+    def add_coordinator(self, serial: str, cord: "NASwebCoordinator") -> None:
+        """Remove NASwebCoordinator to possible notification targets."""
+        self._coordinators[serial] = cord
+        _LOGGER.debug("Added NASwebCoordinator for NASweb[%s]", serial)
+
+    def remove_coordinator(self, serial: str) -> None:
+        """Remove NASwebCoordinator from possible notification targets."""
+        if self._coordinators.pop(serial, None) is None:
```

Comment:
why do we pop with a default argument of None here? Shouldn't we allow it to fail instead?

Reply:
Please don't mark as resolved without replying